Description: In addition to watching /etc/{passwd,shadow,group}, also watch those same files in /var/lib/extrausers/, if they exist.
Author: Michael Terry <mterry@ubuntu.com>

Index: accountsservice-0.6.37/src/daemon.c
===================================================================
--- accountsservice-0.6.37.orig/src/daemon.c
+++ accountsservice-0.6.37/src/daemon.c
@@ -48,6 +48,9 @@
 #define PATH_PASSWD "/etc/passwd"
 #define PATH_SHADOW "/etc/shadow"
 #define PATH_GROUP "/etc/group"
+#define PATH_EXTRAUSERS_PASSWD "/var/lib/extrausers/passwd"
+#define PATH_EXTRAUSERS_SHADOW "/var/lib/extrausers/shadow"
+#define PATH_EXTRAUSERS_GROUP "/var/lib/extrausers/group"
 #define PATH_GDM_CUSTOM "/etc/gdm3/custom.conf"
 #define PATH_LIGHTDM_CONFIG "/etc/lightdm/lightdm.conf"
 
@@ -67,6 +70,9 @@ struct DaemonPrivate {
         GFileMonitor *passwd_monitor;
         GFileMonitor *shadow_monitor;
         GFileMonitor *group_monitor;
+        GFileMonitor *extrausers_passwd_monitor;
+        GFileMonitor *extrausers_shadow_monitor;
+        GFileMonitor *extrausers_group_monitor;
         GFileMonitor *gdm_monitor;
         GFileMonitor *lightdm_monitor;
         GFileMonitor *wtmp_monitor;
@@ -554,6 +560,16 @@ daemon_init (Daemon *daemon)
                                                      PATH_GROUP,
                                                      on_users_monitor_changed);
 
+        daemon->priv->extrausers_passwd_monitor = setup_monitor (daemon,
+                                                                 PATH_EXTRAUSERS_PASSWD,
+                                                                 on_users_monitor_changed);
+        daemon->priv->extrausers_shadow_monitor = setup_monitor (daemon,
+                                                                 PATH_EXTRAUSERS_SHADOW,
+                                                                 on_users_monitor_changed);
+        daemon->priv->extrausers_group_monitor = setup_monitor (daemon,
+                                                                PATH_EXTRAUSERS_GROUP,
+                                                                on_users_monitor_changed);
+
         daemon->priv->wtmp_monitor = setup_monitor (daemon,
                                                     wtmp_helper_get_path_for_monitor (),
                                                     on_users_monitor_changed);
